server/src/main/java/io/littlehorse/common/LHConfig.java

```diff
@@ -451,12 +476,18 @@ public List<ServerListenerConfig> getListeners() {
                     AuthorizationProtocol authProtocol =
                             authMap.get(name) == null ? AuthorizationProtocol.NONE : authMap.get(name);
 
+                    if (authProtocol == AuthorizationProtocol.MTLS && protocol != ListenerProtocol.MTLS) {
```

Comment:
This validates that we only allow MTLS as an authentication mechanism when the listener is set to user MTLS as communication protocol.